### PR TITLE
Feature/appointment regular scheduling frontend

### PR DESCRIPTION
**Pull request by dimitrijevic12**

### CODE REVIEW

File: PatientWebApplication/HealthClinic.DAL/Service/EmployeesScheduleService.cs [hunk 1]
```diff
@@ -61,14 +61,16 @@ private Shift getScheduleShiftForDoctor(DoctorUser doctor, string date, Schedule
 
         public Shift getShiftForDoctorForSpecificDay(string date, DoctorUser doctor)
         {
+            if(doctor == null)
+            {
+                return null;
+            }
             List<Schedule> listOfSchedule = _employeesScheduleRepository.GetAll();
-
             foreach (Schedule schedule in listOfSchedule)
```
Comment (trki293) [inline PatientWebApplication/HealthClinic.DAL/Service/EmployeesScheduleService.cs:69]: Write this in form:
"foreach (Schedule schedule in _employeesScheduleRepository.GetAll())"

File: PatientWebApplication/HealthClinic.DAL/Service/EmployeesScheduleService.cs [hunk 2]
```diff
@@ -61,14 +61,16 @@ private Shift getScheduleShiftForDoctor(DoctorUser doctor, string date, Schedule
...
             {
                 Shift scheduleShiftForDoctor = getScheduleShiftForDoctor(doctor, date, schedule);
                 if (scheduleShiftForDoctor != null) return scheduleShiftForDoctor;
```
Comment (trki293) [inline PatientWebApplication/HealthClinic.DAL/Service/EmployeesScheduleService.cs:72]: Write this in form:
"if (getScheduleShiftForDoctor(doctor, date, schedule) != null) return getScheduleShiftForDoctor(doctor, date, schedule);"

File: PatientWebApplication/PatientWebApplicationIntegrationTests/DoctorAppointmentControllerSetup.cs [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+        public static MyDbContext SetupDbContext()
+        {
+            DoctorUser doctor1 = new DoctorUser(1, "TestDoctorName1", "TestDoctorNameSurname1", "1234", "02/02/2020", "123", "email", "pass", "Grad", 200.0, false, "Cardiology", new List<DoctorNotification>(), "Ordination 2");
+            DoctorUser doctor2 = new DoctorUser(2, "TestDoctorName2", "TestDoctorNameSurname2", "1234", "02/02/2020", "123", "email", "pass", "Grad", 200.0, false, "Pulmonology", new List<DoctorNotification>(), "Ordination 2");
+            DoctorUser doctor3 = new DoctorUser(3, "TestDoctorName3", "TestDoctorNameSurname3", "1234", "02/02/2020", "123", "email", "pass", "Grad", 200.0, false, "Cardiology", new List<DoctorNotification>(), "Ordination 2");
+
+            DbContext.Doctors.Add(doctor1);
+            DbContext.Doctors.Add(doctor2);
+            DbContext.Doctors.Add(doctor3);
```
Comment (trki293) [inline PatientWebApplication/PatientWebApplicationIntegrationTests/DoctorAppointmentControllerSetup.cs:23]: Forward objects to doctors directly as a parameter to the Add () method. For example:
"DbContext.Doctors.Add(new DoctorUser(1, "TestDoctorName1", "TestDoctorNameSurname1", "1234", "02/02/2020", "123", "email", "pass", "Grad", 200.0, false, "Cardiology", new List<DoctorNotification>(), "Ordination 2"));"

File: PatientWebApplication/PatientWebApplicationIntegrationTests/DoctorAppointmentControllerSetup.cs [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+            PatientUser patient1 = new PatientUser(1, "PatientName1", "PatientSurname1", "Female", "1234", "2/2/2020", "123", "2112313", "Alergija", "Grad", false, "email", "pass", false, "Grad2", "Roditelj", null);
+            PatientUser patient2 = new PatientUser(2, "PatientName2", "PatientSurname2", "Female", "1234", "2/2/2020", "123", "2112313", "Alergija", "Grad", false, "email", "pass", false, "Grad2", "Roditelj", null);
+            DbContext.Patients.Add(patient1);
+            DbContext.Patients.Add(patient2);
```
Comment (trki293) [inline PatientWebApplication/PatientWebApplicationIntegrationTests/DoctorAppointmentControllerSetup.cs:28]: Forward patient objects directly as a parameter to the Add () method. For example:
"DbContext.Patients.Add(new PatientUser(1, "PatientName1", "PatientSurname1", "Female", "1234", "2/2/2020", "123", "2112313", "Alergija", "Grad", false, "email", "pass", false, "Grad2", "Roditelj", null));"

File: PatientWebApplication/PatientWebApplicationIntegrationTests/DoctorAppointmentControllerSetup.cs [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+            OperationReferral operationReferral2 = new OperationReferral(2, "Medicine2", "Take medicine until", 3, "Appointment", "comment", 2);
+
+            DbContext.OperationReferrals.Add(operationReferral1);
+            DbContext.OperationReferrals.Add(operationReferral2);
+
+            Operation operation1 = new Operation(1, 2, "03/03/2020", new TimeSpan(0, 14, 0, 0), new TimeSpan(0, 15, 0, 0, 0), 1, "room1");
+            Operation operation2 = new Operation(2, 1, "03/10/2020", new TimeSpan(0, 15, 0, 0), new TimeSpan(0, 15, 15, 0, 0), 2, "room1");
+
+            DbContext.Operations.Add(operation1);
+            DbContext.Operations.Add(operation2);
```
Comment (trki293) [inline PatientWebApplication/PatientWebApplicationIntegrationTests/DoctorAppointmentControllerSetup.cs:82]: Forward objects for classes directly as a parameter to the Add () method. For example:
"DbContext.<ObjectClass>.Add (new <ConstructorForClass>);"

**Review by trki293 — CHANGES_REQUESTED** [review]
Modify the indicated code sequences to meet the principles of Clean code.

**Review by trki293 — APPROVED** [review]
PR is approved.

**Review by Maja0505 — APPROVED** [review]
PR is approved. Good practices and principles of clean code are followed.

**Pull request merged**